src/styles/foundation/_layout.scss

```diff
@@ -1,4 +1,4 @@
-$navigation-width: 240px !default;
+$navigation-width: calc(var(--p-nav-width) / 10 * 1rem) !default;
```

Comment:
I'm not a fan of this code but I think this is the only way to convert it to rems - please correct me if I'm wrong 🙏

Custom properties work at runtime, so using [existing sass utilities](https://github.com/Shopify/polaris-react/blob/master/src/styles/foundation/_utilities.scss#L8) to convert the value to rems like this `rem(var(--p-nav-width))` isn't possible because at the time sass compiles to css `var(--p-nav-width)` doesn't actually hold a value to make maipulations and unit conversions with.

Reply:
Although this will work here, doesn't it break the [calculations](https://github.com/Shopify/polaris-react/blob/688d6b9cecb17eea66277e22d6c75d3699c4d7ad/src/styles/shared/_breakpoints.scss#L58) in the media query helpers below?

I think this has been the major blocker for us trying to do this previously. Since those breakpoint helpers are used in Sass throughout different codebases, we can't move this to a runtime calc without breaking those.

Reply:
Yep, changing these from raw values to properties is a breaking change. [See this other PR that wanted to do a similar thing for reasons](https://github.com/Shopify/polaris-react/pull/3214#issuecomment-686611525)

Reply:
Thank you so much, appreciate the context! 🙏 

---

Would you consider an [alternative solution](https://github.com/Shopify/polaris-react/compare/3212-allow-for-custom-nav-width) to the problem? 

It's removing existing Frame navigation styles and making the navigation be a 100% width of its parent container. It also has some tradeoffs. It's also breaking because consumers would need to add a wrapper with their own width but maybe it could work?

```css
/* consumer's wrapper around navigation */
.Wrapper {
  @include breakpoint-after(nav-min-window-corrected()) {
    width: rem(100px);
  }
}
```